pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/manual.py

```diff
@@ -478,8 +492,10 @@ def _included_thing(self, tag: str, token: Token, tokens: Sequence[Token], i: in
             # we do not set _hlevel_offset=0 because docbook doesn't either.
         else:
             inner = outer
+        in_dir = self._in_dir
         for included, path in fragments:
             try:
+                self._in_dir = (in_dir / path).parent
```

Comment:
I do not really like that this is mutable. Maybe something like `RenderContext` passed to render method would be cleaner.

Alternately, should not we at least restore the `_in_dir` after the include is processed?

Reply:
Ah, you are right. Sorry.